### PR TITLE
Support Validating Webhook Configuration resource in K8s package

### LINKED ISSUE #718: err:creating request for unimplimented resource type:`validatingwebhookconfiguration`
```
Error parsing commandline arguments: error applying 'manifests/cluster-infra/2_ingress-nginx-controller.yaml' err:creating request for unimplimented resource type:validatingwebhookconfiguration
```

We don't have resource implementation for `validatingwebhookconfiguration` in K8s package.

### PR DESCRIPTION
**Pull request by Vandit1604**
fixes: #718 

We aren't supporting `ValidatingWebhookConfiguration` but using it in our cluster-infra manifest files for prombench.

### CODE REVIEW

**Comment by Vandit1604** [comment]
I guess circleci failed because of my committed `go.mod` and `go.sum`. Removed them from the PR to check if that's the case.

**Comment by yomek33** [comment]
I've encountered an error while applying a ValidatingWebhookConfiguration:
```
13:07:44 k8s.go:545: resource created - kind: Job, name: ingress-nginx-admission-patch
13:07:54 provider.go:74: Request for 'running job:ingress-nginx-admission-patch' is done!
13:07:54 k8s.go:703: resource updated - kind: IngressClass, name: nginx
Error parsing commandline arguments: error applying 'manifests/cluster-infra/2_ingress-nginx-controller.yaml' err:resource update failed - kind: ValidatingWebhookConfiguration, name: ingress-nginx-admission: validatingwebhookconfigurations.admissionregistration.k8s.io "ingress-nginx-admission" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update
usage: infra kind resource apply
```
It seems related to missing resourceVersion during an update.

File: pkg/provider/k8s/k8s.go [hunk 1]
```diff
@@ -546,6 +549,44 @@ func (c *K8s) jobApply(resource runtime.Object) error {
...
+		client := c.clt.AdmissionregistrationV1().ValidatingWebhookConfigurations()
+		list, err := client.List(c.ctx, apiMetaV1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "error listing resource : %v, name: %v", kind, req.Name)
+		}
+		var exists bool
+		for _, l := range list.Items {
+			if l.Name == req.Name {
+				exists = true
+				break
```
Comment (yomek33) [inline pkg/provider/k8s/k8s.go:567]: I think adding the following code can resolve the issue

		if req.ResourceVersion == "" {
			req.ResourceVersion = l.ResourceVersion
		}

File: pkg/provider/k8s/k8s.go [hunk 2]
```diff
@@ -546,6 +549,44 @@ func (c *K8s) jobApply(resource runtime.Object) error {
...
+	case "v1":
+		client := c.clt.AdmissionregistrationV1().ValidatingWebhookConfigurations()
+		list, err := client.List(c.ctx, apiMetaV1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "error listing resource : %v, name: %v", kind, req.Name)
+		}
+		var exists bool
+		for _, l := range list.Items {
+			if l.Name == req.Name {
+				exists = true
```
Comment (yomek33) [inline pkg/provider/k8s/k8s.go:574]: I think adding the following code can resolve the issue

	if req.ResourceVersion == "" {
		req.ResourceVersion = l.ResourceVersion
	}

**Comment by Vandit1604** [comment]
Thanks @yomek33 for pointing this out.

**Comment by Vandit1604** [comment]
Updated with changes. Thanks @yomek33

**Review by jesusvazquez — APPROVED** [review]
LGTM very related to your previous change in https://github.com/prometheus/test-infra/pull/716

**Pull request merged**